packages/react-scripts/per-lang-loader/index.js

```diff
@@ -0,0 +1,113 @@
+  this.cacheable && this.cacheable();
+  const options = loaderUtils.getOptions(this) || {};
+
+  if (!options.include) {
+    options.include = ['**/*.json', '**/*.yml', '**/*.yaml'];
+  }
+
+  if (!options.overrides) {
+    options.overrides = [];
+  }
+  const appLocalesDir = path.dirname(this.resource); // this is the absolute path to the index.js in the top level locales dir
+  if (!fs.existsSync(appLocalesDir)) {
+    throw new Error(
+      'Directory does not exist: ' +
+        appLocalesDir +
+        ' for resource: ' +
+        this.resource
+    );
+  }
+  let appResBundle = {};
+  if (options.debug) {
+    console.info(
+      'Bundling locales from ' +
+        appLocalesDir +
+        ' (ordered least specific to most):'
+    );
+  }
+
+  // needs to be ordered in least specialized to most e.g. lib locale -> app locale
+  const moduleLocalesDirs = options.overrides.map(override => {
+    if (path.isAbsolute(override)) {
+      return override;
+    } else {
+      return path.join(appLocalesDir, override);
+    }
+  });
+  moduleLocalesDirs.push(appLocalesDir);
+  moduleLocalesDirs.forEach(localesDir => {
+    // all subdirectories match language codes
+    const langs = enumerateLangs(localesDir);
+    for (let i = 0; i < langs.length; i++) {
+      const lang = langs[i];
+      const resBundle = {};
+      resBundle[lang] = {};
+
+      const fullLangPath = path.join(localesDir, lang);
+      this.addContextDependency(fullLangPath);
+
+      const langFiles = findAll(options.include, fullLangPath);
+      for (let j = 0; j < langFiles.length; j++) {
+        const fullPath = langFiles[j];
+        this.addDependency(fullPath);
+        if (options.debug) {
+          console.info('\t' + fullPath);
+        }
+
+        const fileContent = fs.readFileSync(fullPath);
+        const extname = path.extname(fullPath);
+        let parsedContent;
+        if (extname === '.yaml' || extname === '.yml') {
+          parsedContent = yaml.safeLoad(fileContent);
+        } else {
+          parsedContent = JSON.parse(fileContent);
+        }
+
+        let namespace = 'translation';
+        if (options.basenameAsNamespace || options.relativePathAsNamespace) {
+          let namespaceFilepath;
+          if (options.relativePathAsNamespace) {
+            namespaceFilepath = path.relative(
+              path.join(localesDir, lang),
+              fullPath
+            );
+          } else if (options.basenameAsNamespace) {
+            namespaceFilepath = path.basename(fullPath);
+          }
+          namespace = namespaceFilepath.replace(extname, '').split(path.sep);
+
+          // namespace = [lang].concat(namespaceParts).join(".");
+        }
+        this.addLocales({ lang, content: parsedContent, namespace });
+      }
+    }
+  });
+  const bundle = JSON.stringify(appResBundle);
+  if (options.debug) {
+    console.info('Final locales bundle: \n' + bundle);
+  }
+  // return "module.exports = " + bundle;
+
+  return '';
```

Comment:
is it typical for loaders to return an empty string? I don't really know much about the internals.

Reply:
It can be normal. We'll probably need to do something different to support our case.